api/src/main/java/org/pmiops/workbench/api/DataSetController.java

```diff
@@ -260,9 +260,15 @@ public DomainValuePair apply(DataSetValues dataSetValue) {
 
               queryJobConfiguration = queryJobConfiguration.toBuilder().setQuery(query).build();
 
+              /* Google appengine has a 60 second timeout, we want to make sure this endpoint completes
+               * before that limit is exceeded, or we get a 500 error with the following type:
+               * com.google.apphosting.runtime.HardDeadlineExceededError
+               * See https://cloud.google.com/appengine/articles/deadlineexceedederrors for details
+               */
               TableResult queryResponse =
                   bigQueryService.executeQuery(
-                      bigQueryService.filterBigQueryConfig(queryJobConfiguration), 60000l);
+                      bigQueryService.filterBigQueryConfig(queryJobConfiguration),
+                      50000l / retryLimit + 1);
```

Comment:
why is this dependent on retry limit and not hard coded as 60 sec?

Reply:
Because its the total time limit for the entire API endpoint. So if we're retrying, then the query could easily go over 60 seconds and then we throw a HardDeadlineExceededError